### ISSUE TITLE
[BUG] OpenSearch Serverless Sink does not create index on startup

### ISSUE THREAD

**Issue by graytaylor0** [opensearch-project/data-prepper#4097]
**Describe the bug**
When using an OpenSearch Serverless collection in a pipeline in the `opensearch` sink, the index configured is not created until a document is sent to the pipeline and is indexed into the collection.

**To Reproduce**
1. Create a pipeline with an opensearch serverless collection as a sink
2. Observe the index configured in the sink is not created when starting Data Prepper
3. Send an Event to the pipeline
4. Observe the document in the collection in the specified index

**Expected behavior**
The index should be created on startup like it is for non-serverless opensearch sinks

**Screenshots**

**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu 20.04 LTS]

**Additional context**

**Comment by oeyh** [comment]
I haven't tested it but my understanding is that when using opensearch serverless collection as sink, index_type defaults to `management_disabled`, which is probably the reason pipeline doesn't create index or index template when sink is initialized. Set `index_type` to `custom` would probably make it work.

**Comment by dlvenable** [comment]
@graytaylor0 , Yes, this should work. I think in Data Prepper 3.0 we could fix these defaults. But, this default was chosen because we first added support for Serverless without any support for index management. So changing the default could change existing behavior.
